pkg/agent/core/ngt/service/ngt.go

```diff
@@ -592,44 +592,56 @@ func (n *ngt) CreateIndex(ctx context.Context, poolSize uint32) (err error) {
 			span.End()
 		}
 	}()
-
-	n.cimu.Lock()
-	defer n.cimu.Unlock()
-	if n.IsIndexing() || n.IsSaving() {
-		return nil
-	}
 	ic := n.vq.IVQLen() + n.vq.DVQLen()
 	if ic == 0 {
 		return errors.ErrUncommittedIndexNotFound
 	}
+	err = func() error {
+		ticker := time.NewTicker(time.Millisecond * 100)
+		defer ticker.Stop()
+		// wait for not indexing & not saving
+		for n.IsIndexing() || n.IsSaving() {
```

Comment:
Sorry that if I am wrong, but I think we can put this checking inside the lock without handling by ourself, or we can remove this checking since only 1 goroutine can enter the mutex lock scope. Other than ctx.Done handling, is there any other reason to write this logic here?

Reply:
Since the CreateIndex and SaveIndex are called randomly and asynchronously via the WebAPI, it is necessary to check the Atomic Flag.
Also,`cimu` is a Lock for CreateIndex only, it is not affected by SaveIndex.
Rather, I feel that mutex is inherently more unnecessary.